controllers/suite_test.go

```diff
@@ -77,3 +119,224 @@ var _ = AfterSuite(func() {
 	err := testEnv.Stop()
 	Expect(err).ToNot(HaveOccurred())
 })
+
+type MockSpireService struct {
+	entriesById        map[string]*common.RegistrationEntry
+}
+
+func NewMockSpireService() MockSpireService {
+	return MockSpireService{
+		entriesById:        map[string]*common.RegistrationEntry{},
+	}
+}
+
+func (m MockSpireService) getMatchingEntry(in *common.RegistrationEntry) (*common.RegistrationEntry, error) {
+nextEntry:
+	for _, entry := range m.entriesById {
+		if entry.SpiffeId != in.SpiffeId {
+			continue nextEntry
+		}
+
+		if entry.ParentId != in.ParentId {
+			continue nextEntry
+		}
+		if len(entry.Selectors) != len(in.Selectors) {
+			continue nextEntry
+		}
+		var selectorMap map[string]map[string]bool
+		for _, selector := range entry.Selectors {
+			selectorMap[selector.Type][selector.Value] = true
+		}
+		for _, selector := range in.Selectors {
+			if !selectorMap[selector.Type][selector.Value] {
+				continue nextEntry
+			}
+		}
+		return m.cloneEntry(entry)
+	}
+	return nil, nil
+}
+
+func (m MockSpireService) cloneEntry(in *common.RegistrationEntry) (*common.RegistrationEntry, error) {
+	if in == nil {
+		return nil, nil
+	}
+
+	encoded, err := json.Marshal(in)
+	if err != nil {
+		return nil, err
+	}
+	re := common.RegistrationEntry{}
+	err = json.Unmarshal(encoded, &re)
+	if err != nil {
+		return nil, err
+	}
+	return &re, nil
+}
+
+func (m MockSpireService) CreateEntry(ctx context.Context, in *common.RegistrationEntry, opts ...grpc.CallOption) (*registration.RegistrationEntryID, error) {
+	ceif, err := m.CreateEntryIfNotExists(ctx, in, opts...)
```

Comment:
`ceif` is obscure. How about `e` or `entry`? Function is short so lineage is clear.